Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -113,21 +136,46 @@ ComponentInfo::componentsInSubtree(const size_t componentIndex) const {
 size_t ComponentInfo::size() const { return m_size; }
 
 Eigen::Vector3d ComponentInfo::position(const size_t componentIndex) const {
+  checkNoTimeDependence();
   if (isDetector(componentIndex)) {
     return m_detectorInfo->position(componentIndex);
   }
   const auto rangesIndex = compOffsetIndex(componentIndex);
   return (*m_positions)[rangesIndex];
 }
 
+Eigen::Vector3d
+ComponentInfo::position(const std::pair<size_t, size_t> &index) const {
+
+  const auto componentIndex = index.first;
+  if (isDetector(componentIndex)) {
+    // Time index info must be same between detector and component infos!
+    return m_detectorInfo->position(index);
+  }
+  const auto rangesIndex = compOffsetIndex(componentIndex);
```

Comment:
Would it make sense to merge `compOffsetIndex()` and `linearIndex()`? Do you ever need them separately?